code-name-taran/src/com/spaghettisoft/component/chess/userinput/UserInputValidatorAndConvertor.java

```diff
@@ -0,0 +1,150 @@
+package com.spaghettisoft.component.chess.userinput;
+
+import com.spaghettisoft.component.chess.board.ChessBoard;
+import com.spaghettisoft.component.chess.players.Move;
+import com.spaghettisoft.component.chess.players.Player;
+
+public abstract class UserInputValidatorAndConvertor {

```

Comment:
Class should in general do one thing. The name suggests that it does two things. If you such logic is encapsulated you can rename the class "UserInputProcessor". If the logic is exposed in the interface - better to have two classes : )